File: python/wc.py
import sys

spaces: set = set(map(ord, [' ', '\n', '\t', '\v', '\f', '\r']))

def isspace(b: bytes) -> bool:
	return b not in spaces


word_count: int = 0
in_space: bool = True

for b in sys.stdin.buffer.read():
	if isspace(b):
		in_space = True
	else:
		if in_space:
			word_count += 1
		in_space = False

print(word_count)

File: rust/wc.rs
use std::io;
use std::io::Read;

fn isspace(c: char) -> bool {
    return c == ' ' || c == '\n' || c == '\t' || c == '\x0B' || c == '\x0C' || c == '\r';
}

fn main() {
    let mut word_count: u32 = 0;
    let mut in_space: bool = true;
    for c in io::stdin().lock().bytes() {
        if isspace(c.unwrap().into()) {
            in_space = true;
        } else {
            if in_space {
                word_count += 1;
            }
            in_space = false;
        }
    }
    println!("{:?}", word_count);
}
